packages/applications/ssr/src/components/pages/représentant-légal/changement/corriger/CorrigerChangementReprésentantLégal.form.tsx

```diff
@@ -102,16 +106,25 @@ export const CorrigerChangementReprésentantLégalForm: FC<
           </p>
           <SaisieTypeStep
             contexte="corriger"
-            typeReprésentantLégal={type}
+            typeReprésentantLégal={stateTypeReprésentantLégal}
+            typeSociété={typeSociété}
             validationErrors={validationErrors}
-            onChange={(nouveauType) => setType(nouveauType)}
+            onChange={({ type, typeSociété }) => {
+              setStateTypeReprésentantLégal(type);
+              setTypeSociété(typeSociété);
+            }}
           />
         </div>
       ),
       nextStep: {
         type: 'link',
         name: 'Commencer',
-        disabled: !type || ReprésentantLégal.TypeReprésentantLégal.bind({ type }).estInconnu(),
+        disabled:
+          !stateTypeReprésentantLégal ||
+          (stateTypeReprésentantLégal === 'personne-morale' && typeSociété === 'non renseignée') ||
+          ReprésentantLégal.TypeReprésentantLégal.bind({
+            type: stateTypeReprésentantLégal,
+          }).estInconnu(),
```

Comment:
tant qu'à faire, c'est plus lisible et ca reste typé ;)
```suggestion
          !stateTypeReprésentantLégal ||
          (stateTypeReprésentantLégal === 'personne-morale' && typeSociété === 'non renseignée') ||
stateTypeReprésentantLégal === 'inconnu'
```